src/modes/map/map_mode.cpp

```diff
@@ -1073,6 +1073,19 @@ void MapMode::_DrawMapLayers()
 {
     VideoManager->PushState();
     VideoManager->SetStandardCoordSys();
+
```

Comment:
> Ideally, I think everything (Not just map tiles; map objects and player character too.) should be drawn at the native scale of the map tiles which I think is 32x32 pixels. Then, everything can be scaled up at the end.

Yes, exactly. That would in the end make the engine code cleaner, and much more size agnostic in any case.

> I did some initial tinkering to try and diagnose what would need to happen. However, I ran into some issues. I could never get the map tiles and all the map objects to align in any other way than they are currently. Swapping their coordinate systems and scales around seemed pretty tough to do correctly.

I do think that's because of the overall logic computing the tiles location on screen vs the way it is done for sprites. I guess you already saw where tile offset begin and end are set up, and how sprites are placed on screen?